docs/architecture/adr-027-deterministic-protobuf-serialization.md

```diff
@@ -45,24 +65,38 @@ with the following additions:
    must be omitted
 4. `repeated` fields of scalar numeric types must use
    [packed encoding](https://developers.google.com/protocol-buffers/docs/encoding#packed)
-   by default.
-5. Variant encoding of integers must not be longer than needed.
+5. Varint encoding must not be longer than needed:
+    * No trailing zero bytes (in little endian, i.e. no leading zeroes in big
+      endian). Per rule 3 above, the default value of `0` must be omitted, so
+      this rule does not apply in such cases.
+    * The maximum value for a varint is `FF FF FF FF FF FF FF FF FF 01`.
```

Comment:
Wasn't it `FF FF FF FF FF FF FF FF FF 7F` as stated above? Or do you refer to 64-bit values in this point?

Reply:
These are stricter-than-normal-protobuf restrictions to prevent malleability and thus make the serialization bijective.

Reply:
Ahh, this is to be read as an imperative, no an observation. Then it makes perfect sense.

Maybe this can be clarified using "must" in some way?

Reply:
I see the confusion. Will change!

Reply:
Fixed in eb91c21.

Reply:
Great, thanks!